src/server/qgsfcgiserverresponse.cpp

```diff
@@ -23,15 +23,148 @@
 #include <fcgi_stdio.h>
 #include <QDebug>
 
+#include <unistd.h>
+#include "qgslogger.h"
+
+#ifdef Q_OS_LINUX
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netinet/tcp.h>
+#endif
+
+#ifdef Q_OS_LINUX
 //
-// QgsFcgiServerResponse
+// QgsFCGXStreamData copied from libfcgi FCGX_Stream_Data
 //
+typedef struct QgsFCGXStreamData
+{
+  unsigned char *buff;      /* buffer after alignment */
+  int bufflen;              /* number of bytes buff can store */
+  unsigned char *mBuff;     /* buffer as returned by Malloc */
+  unsigned char *buffStop;  /* reader: last valid byte + 1 of entire buffer.
+                               * stop generally differs from buffStop for
+                               * readers because of record structure.
+                               * writer: buff + bufflen */
+  int type;                 /* reader: FCGI_PARAMS or FCGI_STDIN
+                               * writer: FCGI_STDOUT or FCGI_STDERR */
+  int eorStop;              /* reader: stop stream at end-of-record */
+  int skip;                 /* reader: don't deliver content bytes */
+  int contentLen;           /* reader: bytes of unread content */
+  int paddingLen;           /* reader: bytes of unread padding */
+  int isAnythingWritten;    /* writer: data has been written to ipcFd */
+  int rawWrite;             /* writer: write data without stream headers */
+  FCGX_Request *reqDataPtr; /* request data not specific to one stream */
+} QgsFCGXStreamData;
+#endif
+
+SocketMonitoringThread::SocketMonitoringThread( bool *isResponseFinished, QgsFeedback *feedback )
+  : mIsResponseFinished( isResponseFinished )
+  , mFeedback( feedback )
+  , mIpcFd( -1 )
+{
+  setObjectName( "FCGI socket monitor" );
+  Q_ASSERT( mIsResponseFinished );
+  Q_ASSERT( mFeedback );
+
+#ifdef Q_OS_LINUX
+  if ( FCGI_stdout && FCGI_stdout->fcgx_stream && FCGI_stdout->fcgx_stream->data )
```

Comment:
I think that we should check for the `FCGI_stdin` stream in this case (not `FCGI_stdout`).

Reply:
The fact is the qgis server will need to write the response to the FCGI_stdout. This is why I monitor this stream